common/exception/TypeDBClientException.java

```diff
@@ -85,6 +93,11 @@ private static boolean isPasswordCredentialExpired(StatusRuntimeException status
                 statusRuntimeException.getStatus().getDescription() != null &&
                 statusRuntimeException.getStatus().getDescription().contains(CLUSTER_PASSWORD_CREDENTIAL_EXPIRED_ERROR_CODE);
     }
+
+    private static boolean isUnimplementedMethod(StatusRuntimeException statusRuntimeException) {
+        return statusRuntimeException.getStatus().getCode() == Status.Code.UNIMPLEMENTED;
```

Comment:
This is a sufficient but not necessary condition for a client-server mismatch error. It's also possible that we send a request message with an RPC code that _is_ implemented, but carries a different semantic meaning in this protocol version.

Do we perform some kind of handshake anywhere to verify the protocol version? I notice `OpenReq` now sends a version number. Does that mean the server will decline a connection request from an invalid client?

Reply:
Yep, and it throws basically the equivalent error message if it is a mismatch :) between the two of them we should get pretty good coverage of client-protocol mismatches -> useful errors the users can resolve by themselves.